src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let res = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        res += separator;
+      }
+
+      if (this[i] !== undefined && this[i] !== null) {
+        res += this[i].toString();
```

Comment:
```suggestion
        res += this[i];
```
Элемент и так будет преобразован в строку при конкатенации